pyramid/tests/test_scripts/test_pcreate.py

```diff
@@ -50,11 +50,15 @@ def test_run_no_scaffold_name(self):
             'You must provide at least one scaffold name'))
 
     def test_no_project_name(self):
-        cmd = self._makeOne('-s', 'dummy')
-        result = cmd.run()
-        self.assertEqual(result, 2)
-        out = self.out_.getvalue()
-        self.assertTrue(out.startswith('You must provide a project name'))
+        try:
+            cmd = self._makeOne('-s', 'dummy')
+            result = cmd.run()
+            self.assertEqual(result, 2)
+            out = self.out_.getvalue()
+            self.assertTrue(out.startswith('You must provide a project name'))
+        except SystemExit as e:
+            code = e.args[0]
+            self.assertEqual(code, 2)
```

Comment:
This test shouldn't have needed to be changed - and if it did then it's not properly asserting when the except is not called.